pkg/sources/gitlab/gitlab.go

```diff
@@ -131,6 +133,55 @@ func (s *Source) Init(_ context.Context, name string, jobId, sourceId int64, ver
 	return nil
 }
 
+// Chunks emits chunks of bytes over a channel.
+func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) error {
+	// Start client.
+	apiClient, err := s.newClient()
+	if err != nil {
+		return errors.New(err)
+	}
+	// Get repo within target.
+	repos, errs := s.getRepos()
+	for _, repoErr := range errs {
+		log.WithError(repoErr).Warn("error getting repo")
+	}
+
+	// End early if we had errors getting specified repos but none were validated.
+	if len(errs) > 0 && len(repos) == 0 {
+		return errors.New("All specified repos had validation issues, ending scan")
+	}
+
+	// Get all repos if not specified.
+	if repos == nil {
```

Comment:
**Question:** Should this check `len(repos) == 0` instead? A `nil` slice is technically different than an empty slice, but I'm not sure if we want the empty slice just not scan anything.

Reply:
yep, seems like it.

note: this was just copy pasta code from what we already had. There could likely be additional things i missed.